lib/internal/repl.js

```diff
@@ -84,17 +99,22 @@ function setupHistory(repl, historyPath, ready) {
     if (err) {
       return ready(err);
     }
-
     if (data) {
+      repl.history = data.split(/\n|\r/).slice(-repl.historySize);
```

Comment:
Wouldn't catch a `\n|\r` case, if such a platform exists, so you'd need `/\r\n|\n\r|\r|\n/` but I really have no preference here.